atom/browser/net/url_request_buffer_job.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2013 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_BROWSER_NET_URL_REQUEST_BUFFER_JOB_H_
+#define ATOM_BROWSER_NET_URL_REQUEST_BUFFER_JOB_H_
+
+#include <string>
+
+#include "net/url_request/url_request_simple_job.h"
+#include "atom/common/node_includes.h"
+
+namespace atom {
+
+class URLRequestBufferJob : public net::URLRequestSimpleJob {
+ public:
+  URLRequestBufferJob(net::URLRequest* request,
+                      net::NetworkDelegate* network_delegate,
+                      const std::string& mime_type,
+                      const std::string& charset,
+                      v8::Local<v8::Object> buffer);
+
+// URLRequestSimpleJob:
+int GetData(std::string* mime_type,
```

Comment:
It is not safe to store binary data in `std::string`, you should override `GetRefCountedData` instead of `GetData`.